src/ShopBundle/objects/WebModules/ViewMyOrderDetails/MTShopViewMyOrderDetails.php

```diff
@@ -85,24 +85,39 @@ public function Accept(
             'order' => null,
         );
 
-        if (false === $viewOrderDetailHandler->orderIdBelongsToUser($orderIdRequested, $extranetUserId)) {
-            // this is not our order. access denied
-            $viewData['error'] = true;
-            $viewData['errorCode'] = 'notMyOrderError';
-        }
+        $order = $this->getOrder($orderIdRequested);
 
-        $order = $this->getDbAdapter()->getOrder($orderIdRequested);
         if (null === $order) {
             // this is not our order. access denied
             $viewData['error'] = true;
             $viewData['errorCode'] = 'orderNotFoundError';
+        } else {
+            if (null === $extranetUserId || false === $viewOrderDetailHandler->orderIdBelongsToUser($orderIdRequested, $extranetUserId)) {
+                // this is not our order. access denied
+                $viewData['error'] = true;
+                $viewData['errorCode'] = 'notMyOrderError';
+            }
         }
 
         $viewData['order'] = $order;
 
         $oVisitor->SetMappedValueFromArray($viewData);
     }
 
+    private function getOrder($orderIdRequested): ?\TdbShopOrder
```

Comment:
Does `$orderIdRequested` have a forseeable type (string)?

Reply:
Removed this method. We don't need the additional checks, because the DB adapter class will return null anyway for null or empty string. Then there's only one line left and the method doesn't provide benefit over inlining this one line.